engine/process/main.go

```diff
@@ -98,10 +98,25 @@ func main() {
 
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+
 	err = cmd.Run()
 
 	if err != nil {
 		print.ProcDebug("error running process: %v\n", err)
-		os.Exit(ERunningProc)
+		if cast, ok := err.(*exec.ExitError); ok {
+			if ws, ok := cast.Sys().(syscall.WaitStatus); ok {
+				if ws.Signaled() {
+					// more verbose err message
+					cmd.Stderr.Write([]byte(err.Error()))
+
+					// signal exit code: 128 + signal code
+					os.Exit(128 + int(ws.Signal()))
```

Comment:
error codes are being captured from the `process` command, and luckily the error from `exec.Command` already has a nice explanation of what caused the exit, so I just write that to the command's `stdout` in the case of a signal, such as a segfault